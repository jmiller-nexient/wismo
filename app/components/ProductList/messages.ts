import { defineMessages } from 'react-intl';

export const scope = 'wismo.components.product-list';

export default defineMessages({
  IAC: {
    id: `${scope}.IAC`,
    defaultMessage: 'Interuptable Air Conditioning',
  },
  RES_ELECTRIC: {
    id: `${scope}.RES_ELECTRIC`,
    defaultMessage: 'Residential Electric Service',
  },
  RES_GAS: {
    id: `${scope}.RES_GAS`,
    defaultMessage: 'Residential Gas Heating',
  },
});
