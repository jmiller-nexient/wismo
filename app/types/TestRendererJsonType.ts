import { ReactTestRendererJSON } from 'react-test-renderer';

type TestRendererJsonType = ReactTestRendererJSON | null;

export default TestRendererJsonType;
