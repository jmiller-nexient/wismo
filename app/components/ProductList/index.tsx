import * as React from 'react';
import { FormattedMessage } from 'react-intl';

import IProduct from 'types/product';

import messages from './messages';

interface IProductListProps {
  products: (IProduct[] | null);
}

const ProductList: React.FC<IProductListProps> = (props: IProductListProps): null | React.ReactElement => {
  const { products } = props;

  if (!products) {
    return null;
  }

  const productList = products.map((product, index) => {
    return (
      <React.Fragment key={product.productType}>
        <FormattedMessage {...messages[`${product.productType}`]} />
        {index < products.length - 1 ? ', ' : null}
      </React.Fragment>
    );
  });

  return (
    <>
      {productList}
    </>
  );
};

export default ProductList;
