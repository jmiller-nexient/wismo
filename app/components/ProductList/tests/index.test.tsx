import React from 'react';
import renderer from 'react-test-renderer';

import ProductList from 'components/ProductList';

import IProduct from 'types/product';
import TestRendererJsonType from 'types/TestRendererJsonType';

const emptyList: IProduct[] =[];
const oneItemList: IProduct[] = [
  {
    productType: 'IAC',
    serviceDate: '2020-03-21'
  }
];
const threeItemList: IProduct[] = [
  {
    productType: 'IAC',
    serviceDate: '2020-03-21'
  },
  {
    productType: 'RES_GAS',
    serviceDate: '2020-03-15'
  },
  {
    productType: 'RES_ELECTRIC',
    serviceDate: '04-22-2020'
  }
];

describe('<ProductList />', () => {
  it('should render correctly when a product list with multiple items is passed', () => {
    const tree: TestRendererJsonType = renderer.create(
      <ProductList products={threeItemList} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when a product list with a single item is passed', () => {
    const tree: TestRendererJsonType = renderer.create(
      <ProductList products={oneItemList} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when product list is empty', () => {
    const tree: TestRendererJsonType = renderer.create(
      <ProductList products={emptyList}/>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
