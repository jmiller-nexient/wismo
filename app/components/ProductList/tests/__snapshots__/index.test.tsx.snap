// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductList /> should render correctly when a product list with a single item is passed 1`] = `"Interuptable Air Conditioning"`;

exports[`<ProductList /> should render correctly when a product list with multiple items is passed 1`] = `
Array [
  "Interuptable Air Conditioning",
  ", ",
  "Residential Gas Heating",
  ", ",
  "Residential Electric Service",
]
`;

exports[`<ProductList /> should render correctly when product list is empty 1`] = `null`;
